packages/@n8n/task-runner/src/js-task-runner/__tests__/js-task-runner.test.ts

```diff
@@ -1412,5 +1412,28 @@ describe('JsTaskRunner', () => {
 			expect(outcome.result).toEqual([wrapIntoJson({ prototypeChanged: false })]);
 			checkPrototypeIntact();
 		});
+
+		test('should freeze luxon prototypes', async () => {
+			const outcome = await executeForAllItems({
+				code: `
+				[DateTime, Interval, Duration]
+						.forEach(constructor => {
+								constructor.prototype.maliciousKey = 'value';
+						});
+
+				return []
+				`,
+				inputItems: [{ a: 1 }],
+			});
+
+			expect(outcome.result).toEqual([]);
```

Comment:
Do we need this assertion? Is this for checking that it doesn't throw?

Reply:
I left it there to check that the code was executed as expected